xapian-core/include/xapian/weight.h

```diff
@@ -409,6 +409,11 @@ class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
     /// The factor to multiply with the weight.
     double factor;
 
+    /// Parameters slope and delta in the Piv+ normalization weighting formula.
+    /// Default values: 0.2 and 1.0 respectively.
+    double param_slope = 0.2;
+    double param_delta = 1.0;
+
     TfIdfWeight * clone() const;
```

Comment:
@ojwb Sorry about the confusion. I think I was confusing delta used here with the one used with BM25+ and thus set its default value to be 1.0 but for slope, the value 0.2 came from table 10 and conclusions from the text in the two paragraphs below it. Overall there's no particular default value for piv+ delta, not even in the summary section (7.6). 

We actually want the slope to assume values between 0.01 to 0.25 and delta between 0.0 to 1.5 (or 0.1 to 1.5 so that both params assume only positive values) according to the paper.

Should we try to reject `P` normalisations when params values are not specified as you suggested above? We can achieve this by removing `P` normalisation character from the invalid normalisation string check in `TfIdfWeight::TfIdfWeight(const std::string &normals)` as:

```c++
if (normalizations.length() != 3 ||
	!strchr("nbsl", normalizations[0]) ||
	!strchr("ntpfs", normalizations[1]) ||
	!strchr("n", normalizations[2]))
	throw Xapian::InvalidArgumentError("Normalization string is invalid");
``` 

and add a check for the situation when `para_slope` and `param_delta` values are set zero or negative which assigns a positive value to them (not sure which values to assign) in the definition of `TfIdfWeight::TfIdfWeight(const std::string &normals, double slope, double delta)`.